htseq2multx.pl

```diff
@@ -459,15 +467,25 @@ sub runFastqMultx
           }
       }
 
-    if($CHILD_ERROR || $make_error_fatal)
+    waitpid($pid, 0);
+    my $exit_code = $CHILD_ERROR >> 8;
+
+    if($exit_code)
+      {
+        print STDERR ('ERROR: fastq-multx command failed with a non-zero ',
+                      "exit code [$exit_code]",
+                      ($OS_ERROR =~ /./ ? " and error: [$OS_ERROR]" : ''),
+                      ":\n$command\n");
+
+        #Exit this script indicating an error with the fastq-multx executable
+        exit($FQMX_ERROR_CODE);
+      }
+    #Or exit non-zero if one of the make_error_fatal patterns matched the
+    #STDERR output and STDOUT has no tab characters.
```

Comment:
NB: It's not clear to me why a tab character overrides the `$make_error_fatal` value. I'm sure it's correct, just not clear from the comments, imho.

Reply:
Yeah, I also struggled over the tab and decided to err on the side of keeping the behavior consistent. I think it's entirely possible that I'd added the tab requirement to mitigate cases where `$CHILD_ERROR` was non-zero in the original code yet the output was complete.  I wish I'd documented why I put that tab-case there.

I suspect that I had mis-remembered how `IO::Pipe::Producer` exits and that the child could have been exiting non-zero for a legitimate reason despite it looking successful, but I feel like I remember finding code in fastq-multx source where it was exiting non-0 for an inane reason.

It could be that I have fixed things such that it won't inappropriately exit non-0, in which case the tab check is unnecessary. Unfortunately, the only way to find out would be in the wild. So what do you think? Should I remove the tab check and risk a job failing so that I can improve this code or err on the side of consistency?

Reply:
I wouldn't complicate this change with too many things. Let's make that a separate issue if it needs to be so we have time to test, etc.

Reply:
#10